libyul/optimiser/Metrics.cpp

```diff
@@ -70,8 +70,13 @@ size_t CodeWeights::costOf(Expression const& _expression) const
 		return functionCallCost;
 	else if (holds_alternative<Identifier>(_expression))
 		return identifierCost;
-	else if (holds_alternative<Literal>(_expression))
-		return literalCost;
+	else if (Literal const* literal = get_if<Literal>(&_expression))
+	{
+		if (literal->kind == LiteralKind::Number && literal->value == "0"_yulstring)
```

Comment:
Shouldn't this properly check if the `u256` value is zero? This would make ``0x0`` be more expensive than ``0`` now, wouldn't it?